jbpm-bpmn2/src/main/java/org/jbpm/bpmn2/xml/StartEventHandler.java

```diff
@@ -102,6 +103,15 @@ protected void handleNode(final Node node, final Element element, final String u
                 break;
             } else if ("signalEventDefinition".equals(nodeName)) {
                 String type = ((Element) xmlNode).getAttribute("signalRef");
+                Map<String, Signal> signals = (Map<String, Signal>) ((ProcessBuildData) parser.getData()).getMetaData("Signals");
+                
+                if (signals != null && signals.containsKey(type)) {
+                    Signal signal = signals.get(type);                      
+                    type = signal.getName();
+                    if (type == null) {
+                        throw new IllegalArgumentException("Signal definition must have a name attribute");
```

Comment:
Argh, okay. Leave them here. I'll move them to the RuleFlowProcessValidator class later.